homeassistant/components/osoenergy/__init__.py

```diff
@@ -70,13 +77,18 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     return unload_ok
 
 
-class OSOEnergyEntity(Entity, Generic[_T]):
+class OSOEnergyEntity(Entity, Generic[_OSOEnergyT]):
     """Initiate OSO Energy Base Class."""
 
     _attr_has_entity_name = True
 
-    def __init__(self, osoenergy: OSOEnergy, osoenergy_device: _T) -> None:
+    def __init__(self, osoenergy: OSOEnergy, device: _OSOEnergyT) -> None:
         """Initialize the instance."""
         self.osoenergy = osoenergy
-        self.device = osoenergy_device
-        self._attr_unique_id = osoenergy_device.device_id
+        self.device = device
```

Comment:
With the addition of the sensor entity, `device` is no longer a good name, see the comments on `sensor.py`.